### PR TITLE
{ARM} ts: show command bug fixed, improved error handling for export

### PR DESCRIPTION
**Pull request by detienne20**
**Description**

This PR addresses two customer reported issues. 

For the first (https://github.com/Azure/azure-cli/issues/16736) the error handling has been improved by throwing an incorrect usage error when version is not specified when calling ts export. 

    Testing: 
     
     az ts export -g resourceGroup --name templatespec name --output-folder
     az ts export -s templatespecID 
     
     Both commands should fail with " a template spec version must be specified for 
     export."

For the second ( https://github.com/Azure/azure-cli/issues/16801#issuecomment-773264303) az ts show now reports existing template spec versions instead of returning null. 

   Testing: 

      1. Created a template spec and template spec version(s). 
     
      2. Retrieve a template spec  via:
          ts show --template-spec resourceID 
          ts show -g resourceGroup --name TemplateSpecName 

      3. Returned object should have the name of the template spec versions in the versions 
         property.

### CODE REVIEW

**Comment by yonzhan** [comment]
ARM

File: src/azure-cli/azure/cli/command_modules/resource/custom.py [hunk 1]
```diff
@@ -1935,7 +1938,9 @@ def export_template_spec(cmd, output_folder, resource_group_name=None, name=None
         version = id_parts.get('resource_name')
         if version == name:
             version = None
-    exported_template = rcf.template_spec_versions.get(resource_group_name, name, version) if version else rcf.template_specs.get(resource_group_name, name)
+    if not version:
+        raise IncorrectUsageError('a template spec version must be specified for export')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/resource/custom.py:1942]: Please use a more friendly voice
```suggestion
        raise IncorrectUsageError('Please specify the template spec version for export')
```

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py [hunk 2]
```diff
@@ -1017,8 +1024,17 @@ def test_export_template_spec(self, resource_group, resource_group_location):
...
 
+    @ResourceGroupPreparer(name_prefix='cli_test_export_template_spec', location="westus")
+    def test_template_spec_export_error_handling(self, resource_group, resource_group_location):
+        self.kwargs.update({
+            'template_spec_name': 'CLITestTemplateSpecExport',
+            'output_folder': os.path.dirname(os.path.realpath(__file__)).replace('\\', '\\\\')
+        })
+        with self.assertRaises(IncorrectUsageError) as err:
+            self.cmd('ts export -g {rg} --name {template_spec_name} --output-folder {output_folder}')
+            self.assertTrue('A template spec version must be specified for export.' in str(err.exception))
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:1037]:
```suggestion
            self.assertTrue('Please specify the template spec version for export' in str(err.exception))
```

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**